### ISSUE TITLE
No per-file :cljc :features, so mixed portable/platform-specific projects cannot be linted correctly

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2922]
`:cljc {:features ...}` is global, so a project containing both portable and platform-specific `.cljc` cannot be linted correctly. There is no invocation-level workaround, because in editor mode clj-kondo runs off `.clj-kondo/config.edn` with no flags.

## Repro

```
src/portable.cljc   (ns portable) (defn f [x] x)
src/jsonly.cljc     (ns jsonly)   (defn g [] (js/Set.))
src/consumer.clj    (ns consumer (:require [portable])) (defn h [] (portable/f 1))
```

With `{:cljc {:features [:cljs]}}`, the portable namespace does not exist as far as the `.clj` consumer is concerned:

```
$ clj-kondo --lint src
src/consumer.clj:2:13: warning: Unresolved var: portable/f
```

With `{:cljc {:features [:clj :cljs]}}` that resolves, but the JS-only file is now read as clj:

```
$ clj-kondo --lint src
src/jsonly.cljc:2:13: warning: Unresolved namespace js. Are you missing a require? [clj]
```

Neither setting is right for the project as a whole. Every editor buffer shows false positives in one half or the other.

## Where this came up

reagami has JS-only namespaces (`reagami.core` and friends, full of `js/`, `js-in`, `js-delete`, `undefined?`) alongside a genuinely portable one that runs on the JVM, babashka, CLJS and squint, plus a babashka-only `.clj` that requires it.

`[:cljs]` gives false `Unresolved var` on every call from the `.clj` file into the portable namespace. `[:clj :cljs]` fixes those and produces 19 errors and 28 warnings elsewhere: `Uninitialized var` for each `#?(:squint ... :cljs ...)` def with no `:clj` branch, `Unresolved symbol` for `js-in`, `js-delete`, `undefined?`, `object?`, and one `let binding vector requires even number of forms` where a reader conditional had no `:clj` branch.

## What does not work

Three mechanisms look like they should scope it. All three are ignored and the `[clj]` warning is unchanged:

```clojure
{:cljc {:features [:clj :cljs]}
 :config-in-ns {jsonly {:cljc {:features [:cljs]}}}}
```

```clojure
{:cljc {:features [:clj :cljs]}
 :ns-groups [{:name g :pattern "jsonly"}]
 :config-in-ns {g {:cljc {:features [:cljs]}}}}
```

```clojure
(ns ^{:clj-kondo/config '{:cljc {:features [:cljs]}}} jsonly)
```

That is presumably structural, since `:features` gates reading and the namespace is not known until after the read. If so, a path-keyed override rather than a namespace-keyed one would be the shape that can work, something like:

```clojure
{:cljc {:features [:clj :cljs]
        :overrides [{:paths ["src/reagami/core.cljc"] :features [:cljs]}]}}
```

The only workaround today is `:ns-groups` plus `:config-in-ns` to disable `unresolved-symbol`, `unresolved-namespace`, `uninitialized-var`, `invalid-arity` and `unused-binding` on the platform-specific namespaces. That means hand-listing every affected namespace and losing those linters on the cljs side too, which is worse than the false positives.

## Version

`clj-kondo v2026.05.26-SNAPSHOT`
